torch/linalg/__init__.py

```diff
@@ -140,3 +140,43 @@
     >>> LA.norm(m[0, :, :]), LA.norm(m[1, :, :])
     (tensor(3.7417), tensor(11.2250))
 """)
+
+tensorinv = _add_docstr(_linalg.linalg_tensorinv, r"""
+linalg.tensorinv(input, ind=2, *, out=None) -> Tensor
+
+Computes a tensor ``input_inv`` such that ``tensordot(input_inv, input, ind) == I_n``,
+where ``I_n`` is the n-dimensional identity tensor and ``n`` is equal to ``input.ndim``.
+The resulting tensor ``input_inv`` has the shape equal to ``input.shape[ind:] + input.shape[:ind]``.
+
+Supports real and, only on the CPU, complex inputs.
+
+.. note:: If :attr:`input` is not a 'square' tensor, meaning it does not satisfy the requirement
```

Comment:
I would remove the idea of input being "square" since "square" = nxn matrix. I think this can just state the requirement plainly. 

"If :attr:\`input\` is not invertible or \`\`prod(input.shape[ind:]) != prod(input.shape[:ind])\`\` then a RuntimeError is thrown." 

Note that the first part of this sentence is only recently true on CUDA.